packages/core/src/workers/password-encryption-worker.ts

```diff
@@ -0,0 +1,36 @@
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { findUp } from 'find-up';
+import { Tinypool } from 'tinypool';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Use findUp to locate the build directory
+const buildDirectory = await findUp('build', { type: 'directory', cwd: __dirname });
```

Comment:
better to stop at the first `package.json` file?

options:

1. use `package-up`, and replace `find-up` with `find-up-simple`
2. implement stop logic using [findUpStop](https://www.npmjs.com/package/find-up#findupstop), and replace `find-up` with `find-up-simple` later

Reply:
How about this? Use the [pkg-dir](https://github.com/sindresorhus/pkg-dir) instead, locate the package directory (should auto stop at package.json), and directly build the path.

Reply:
yes, but we still needs to replace `find-up` with `find-up-simple` to reduce the number of dependencies. you can do it in a separate pull request.

Reply:
👌